### PR TITLE
Refactor integration test to verify Lambda and Spring integration with a sample controller

### PR DESCRIPTION
**Pull request by wbingli**
#### Summary
As for our integration test,  we want to use the integration test to focus on following:
 - Lambda to Spring mapping 
 - Spring context wiring 
 - Exception handling

As for the API logic itself, we will test each API in the controller unit test.

In this PR, it removes the per API integration and replace with a sample controller. It will verify the Lambda request to the Spring controller mapping, such as Request mapping, response etc.  

It also keep the whole Spring context scan as the same as production runtime to verify Spring context wiring.

As for the next step, we will use this framework to test the exception handling. 

#### Implementation details
Notice that I don't use lombok for the `SampleResponse` bean. Because lombok is a compileOnly dependency. Instead of adding extra dependency,  the basic constructor and getter/setter makes more sense. 

#### Testing
`gradle check`

#### Licensing

This contribution is under the terms of the Apache 2.0 License: Yes

### CODE REVIEW

File: frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/IntegrationTestBase.java [hunk 1]
```diff
@@ -40,28 +36,9 @@
 )
 public abstract class IntegrationTestBase {
```
Comment (wjbuys) [inline frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/IntegrationTestBase.java:40]: Now that we'll only have the one integration test, is it still worth having the base class?

File: frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/IntegrationTestConfig.java [hunk 2]
```diff
@@ -48,6 +49,16 @@
...
   }
 
+  @Bean
+  public SampleController sampleController() {
+    return new SampleController();
+  }
+
+  @Bean
+  public Checker sampleControllerChecker() {
+    return mock(Checker.class);
```
Comment (wjbuys) [inline frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/IntegrationTestConfig.java:59]: Did you figure out if this actually resets mocks after every test?

File: frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/SampleController.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    return checker.handle(path, query);
+  }
+
+  static class SampleResponse {
+    private String result;
+
+    public SampleResponse() {}
+
+    public SampleResponse(final String result) {
+      this.result = result;
```
Comment (wjbuys) [inline frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/SampleController.java]: I'd be totally fine with including lombok as a `testCompileOnly` dependency, I've been doing that for the JSON-RPC tests.

**Review by wjbuys — APPROVED** [review]
Nice and simple, I like it. Feel free ship this and address comments in your next PR.

File: frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/IntegrationTestBase.java (hunk 1 above)
Comment (wbingli) [reply frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/IntegrationTestBase.java:40]: I plan to use this one for exception testing as well.  Will evaluate if it's still needed then.

File: frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/SampleController.java (hunk 3 above)
Comment (wbingli) [reply frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/SampleController.java]: OK, will update.

File: frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/IntegrationTestConfig.java (hunk 2 above)
Comment (wbingli) [reply frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/IntegrationTestConfig.java:59]: Nope, will still need a reset for the mock once we have more test cases.  Will update to address this issue.

**Pull request merged**